iati/core/tests/test_data.py

```diff
@@ -169,3 +169,35 @@ def test_dataset_xml_tree_assignment_invalid_value(self, dataset_initialised, in
         else:  # pragma: no cover
             # a TypeError should be raised when creating without a tree
             assert False
+
+    def test_instantiate_dataset_from_string(self):
+        """Test that a dataset instantiated directly from a string (rather than a file) correctly creates an iati.core.data.Dataset and the input data is contained within the object."""
+        xml = """<?xml version="1.0"?>
+        <iati-activities version="xx">
+          <iati-activity>
+             <iati-identifier></iati-identifier>
+         </iati-activity>
+        </iati-activities>"""
+
+        dataset = iati.core.data.Dataset(xml)
+
+        assert isinstance(dataset, iati.core.data.Dataset)
+        assert dataset.xml_str == xml
+
+    @pytest.mark.parametrize("encoding", ["UTF-8", "utf-8", "UTF-16", "utf-16"])
+    def test_instantiate_dataset_from_string_with_encoding(self, encoding):
+        """Test that an encoded dataset instantiated directly from a string (rather than a file) correctly creates an iati.core.data.Dataset and the input data is contained within the object.
+        Only UTF-8 or UTF-16 are specified as these are those strongly recommended in IATI Guidance: http://iatistandard.org/202/guidance/how-to-publish/select-data-to-publish/
+        """
+        xml = """<?xml version="1.0" encoding="{}"?>
+        <iati-activities version="xx">
+          <iati-activity>
+             <iati-identifier></iati-identifier>
+         </iati-activity>
+        </iati-activities>""".format(encoding)
+        xml_encoded = xml.encode(encoding)  # Encode the whole string in line with the specified encoding
+
+        dataset = iati.core.data.Dataset(xml_encoded)
+
+        assert isinstance(dataset, iati.core.data.Dataset)
+        assert dataset.xml_str == xml_encoded
```

Comment:
Had a look at this, and have concluded that this might not be that strightforward, as to decode a string – to say `utf-8` – you need to know what encoding it is in. An alternative might be to store store the XML string as the output of `etree.tostring`, which would let `lxml` deal with all the encoding business.